tensorpack/dataflow/common.py

```diff
@@ -684,61 +690,97 @@ def __init__(self, ds, num=1, label=None, name=None):
         else:
             self.name = name
         self.cnt = 0
+        self.max_depth = max_depth
+        self.max_list = max_list
 
-    def _analyze_input_data(self, entry, k, depth=1):
+    def _analyze_input_data(self, entry, k, depth=1, max_depth=3, max_list=3):
         """
         Gather useful debug information from a datapoint.
 
         Args:
             entry: the datapoint component
             k (int): index of this compoennt in current datapoint
             depth (int, optional): recursion depth
-
-        Todo:
-            * call this recursively and stop when depth>n for some n if an element is a list
+            max_depth (int, optional): stop output when to deep recursion in sub elements
+            max_list (int, optional): stop output when too sub elements
```

Comment:
Not used anywhere. And docs should be with \_\_init\_\_

Reply:
FixedSizeData has docs in __init__, JoinData has docs in class. This is not consistent?!

Reply:
Yes that's a bit inconsistent.. but at least arguments should be with the function.

Reply:
Just to not being confused in 61e02bc the arguments are documented? Could you somehow document how to locally build the documentation such that it can be fixed without pull-requests?

Reply:
Now are documented. I would remove the duplicated docs from here, or mark it "the same as init".
Does the readme in https://github.com/ppwwyyxx/tensorpack/tree/master/docs work?

Reply:
Oh silly me. I didn't saw the docs dir. I tested it the documentation looks ok. I would prefer keeping the example in the class scope like PrintData currently does.